helix-term/src/commands/typed.rs

```diff
@@ -2167,6 +2169,64 @@ fn reset_diff_change(
     Ok(())
 }
 
+fn rename_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+    ensure!(args.len() == 1, ":rename takes one argument");
+    let new_name = args.first().unwrap();
+
+    let (_, doc) = current!(cx.editor);
+    let path = doc
+        .relative_path()
+        .ok_or_else(|| anyhow!("Buffer not saved, use :write"))?
+        .canonicalize()
+        .map_err(|_| anyhow!("Could not get absolute path of the document"))?;
+    let mut path_new = path.clone();
+    path_new.set_file_name(OsStr::new(new_name.as_ref()));
+    if path_new.exists() {
+        bail!("This file already exists");
+    }
+
+    if let Err(e) = std::fs::rename(&path, &path_new) {
+        bail!("Could not rename file, error: {}", e);
+    }
+    let (_, doc) = current!(cx.editor);
+    doc.set_path(Some(path_new.as_path()))
+        .map_err(|_| anyhow!("File renamed, but could not set path of the document"))?;
+
+    if let Some(lsp_client) = doc.language_server() {
+        if let Ok(old_uri_str) = Url::from_file_path(&path) {
+            let old_uri = old_uri_str.to_string();
+            if let Ok(new_uri_str) = Url::from_file_path(&path_new) {
+                let new_uri = new_uri_str.to_string();
+                let files = vec![lsp::FileRename { old_uri, new_uri }];
+                match helix_lsp::block_on(lsp_client.will_rename_files(&files)) {
+                    Ok(edit) => {
+                        if apply_workspace_edit(cx.editor, helix_lsp::OffsetEncoding::Utf8, &edit)
+                            .is_err()
+                        {
+                            log::error!(":rename command failed to apply edits")
```

Comment:
we already show an error message herer you don't need to display it. The returned result is just forcases where we need to communicate to the LSP that the edit failed (not here)